.github/workflows/php-quality-checks.yml

```diff
@@ -0,0 +1,607 @@
+            echo "✅ PHPMD configuration found"
+            echo "has-phpmd-config=true" >> $GITHUB_OUTPUT
+          else
+            echo "ℹ️ No PHPMD configuration found, will use defaults"
+            echo "has-phpmd-config=false" >> $GITHUB_OUTPUT
+          fi
+
+  # Install Composer dependencies with caching
+  install:
+    name: 📦 Install Dependencies
+    needs: prepare
+    if: needs.prepare.outputs.has-composer == 'true'
+    runs-on: ubuntu-latest
+    container:
+      image: ${{ needs.prepare.outputs.php-container }}
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Get Composer cache directory
+        id: composer-cache
+        run: |
+          echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
+
+      - name: Cache Composer dependencies
+        uses: actions/cache@v4
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-php${{ inputs.php-version }}-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-composer-php${{ inputs.php-version }}-
+            ${{ runner.os }}-composer-
+
+      - name: Install system dependencies
+        run: |
+          apk add --no-cache git unzip
+
+      - name: Install Composer
+        run: |
+          curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
+          composer --version
+
+      - name: Validate composer.json and composer.lock
+        run: |
+          composer validate --strict ${{ inputs.debug && '--verbose' || '' }}
+
+      - name: Install dependencies
+        run: |
+          php -ini | grep memory_limit
+          php -d memory_limit=${{ inputs.memory-limit }} /usr/local/bin/composer install \
+            --no-progress \
+            --no-suggest \
+            --no-interaction \
+            --optimize-autoloader \
+            --classmap-authoritative \
+            ${{ inputs.composer-args }} \
+            ${{ inputs.debug && '--verbose' || '' }}
+
+      # Archive vendor directory for other jobs
+      - name: Archive vendor directory
+        run: tar -czf vendor.tar.gz vendor/
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: vendor-php${{ inputs.php-version }}
+          path: vendor.tar.gz
+          retention-days: 1
+
+  # Composer security audit
+  security-audit:
+    name: 🔒 Security Audit
+    needs: [prepare, install]
+    if: needs.prepare.outputs.has-composer == 'true'
+    runs-on: ubuntu-latest
+    container:
+      image: ${{ needs.prepare.outputs.php-container }}
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Install Composer
+        run: |
+          apk add --no-cache git unzip
+          curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
+
+      - uses: actions/download-artifact@v4
```

Comment:
Composer should also be a downloadable artifact to save downloading from composer over and over. It should be cached at the repository level rather than per workflow run